README.md

```diff
@@ -20,28 +20,65 @@ A simple usage example:
   import 'package:ordered_set/ordered_set.dart';
 
   main() {
-    final items = OrderedSet<int>();
+    final items = OrderedSet.simple<num, int>();
     items.add(2);
     items.add(1);
     print(items.toList()); // [1, 2]
   }
 ```
 
+But it can accept multiple items with the same priority:
+
+```dart
+  import 'package:ordered_set/ordered_set.dart';
+
+  main() {
+    final items = OrderedSet.mapping<String, Person>((p) => p.name);
+    items.add(Person('Alice', 'Engineering'));
+    items.add(Person('Bob', 'Accounting'));
+    items.add(Person('Alice', 'Marketing'));
+    print(items.toList()); // [Alice, Alice, Bob]
+  }
+```
+
 ## Comparing
 
-In order to assist the creation of OrderedSet's, there is a Comparing class to easily create Comparables:
+In order to assist the creation of `Comparator`s, the `Comparing` class can be used:
 
 ```dart
   // sort by name length
-  final people = OrderedSet<Person>(Comparing.on((p) => p.name.length));
+  final people = OrderedSet.comparing<Person>(Comparing.on((p) => p.name.length));
 
   // sort by name desc
-  final people = OrderedSet<Person>(Comparing.reverse(Comparing.on((p) => p.name)));
+  final people = OrderedSet.comparing<Person>(Comparing.reverse(Comparing.on((p) => p.name)));
 
   // sort by role and then by name
-  final people = OrderedSet<Person>(Comparing.join([(p) => p.role, (p) => p.name]));
+  final people = OrderedSet.comparing<Person>(Comparing.join([(p) => p.role, (p) => p.name]));
 ```
 
+Note that you could instead just create a `MappingOrderedSet` instead:
+
+```dart
+  final people = OrderedSet.mapping<num, Person>((p) => p.name.length);
+  // ...
+```
+
+## Mapping vs Comparing vs Queryable
+
+There are three main implementations of the `OrderedSet` interface:
+
+* `ComparingOrderedSet`: the simplest implementation, takes in a `Comparator` and does not cache priorities. It uses Dart's `SplayTreeSet` as a backing implementation.
+* `MappingOrderedSet`: a slightly more advanced implementation that takes in a mapper function (maps elements to their priorities) and caches them. It uses Dart's `SplayTreeMap` as a backing implementation.
+* `QueryableOrderedSet`: a simple wrapper over either `OrderedSet` that allows for O(1) type queries; if you find yourself doing `.whereType<T>()` a lot, you should consider using this.
+
+In order to create an `OrderedSet`, however, you can just use the static methods on the interface itself:
+
+* `comparing<E>([comparator])`: creates a `ComparingOrderedSet` with the given `Comparator`.
```

Comment:
I think it is even clearer here to write `OrderedSer.`, even though it is mentioned in the sentence before.